elastic/apm-server
# publish: implement BatchProcessor; remove tracing

## Motivation/summary

- Modify `publish.Publisher` so that it implements `model.BatchProcessor` directly.
- Remove the tracing instrumentation from publisher. We don't use and don't support libbeat processors, so this is not useful.

## Checklist

~- [ ] Documentation has been updated~

## How to test these changes

1. Enable instrumentation
2. Send some events to apm-server
3. Check that no "Publisher" type transactions are traced

## Related issues

Related to https://github.com/elastic/apm-server/issues/6002
